delegates.rb

```diff
@@ -83,93 +84,79 @@ def redirect(options = {})
   def authorize(options = {})
     logger = Java::edu.illinois.library.cantaloupe.script.Logger
    
-    u_file_access = ['10.128.99.55','10.128.1.167','10.224.6.10','10.128.99.167','10.128.98.50','10.224.6.26','10.224.6.35','172.16.1.94', '66.234.38.35']
+    g_file_access = ['10.128.99.55','10.128.1.167','10.224.6.10','10.128.99.167','10.128.98.50','10.224.6.26','10.224.6.35','172.16.1.94', '66.234.38.35']
     #'65.88.88.115'
     remote_ip = context['request_headers']['X-Forwarded-For']
     logger.debug("CONTEXT HASH: #{context}")
+    logger.debug("REQUEST HEADERS: #{context['request_headers']}")
+    logger.debug("CLIENT_IP: #{context['client_ip']}")
     logger.debug("IP ADDRESS: #{remote_ip}")
     logger.debug("REQUEST URI: #{context['request_uri']}")
-    # set type to variable since it will be referenced more frequently in future work
-    type = context['request_uri'].split('=')[1]
-    logger.debug("type: #{type}")
-    if context['request_uri'] =~ /ufile=true/
-      logger.debug("UFILE ACCESS")
-      if u_file_access.include?(remote_ip) || remote_ip =~ /^63.147.60./
+    split_image_url = context['request_uri'].split('/') # ie. ["http:", "", "localhost:8182", "iiif", "2", "1992268", "full", "!150,150", "0", "default.jpg"]
+    type = split_image_url[7]=="full" ? split_image_url[7] : derivative_type(context['resulting_size'])
+    logger.debug("TYPE: #{type}")
+    if type == "full"
+      logger.debug("GFILE ACCESS")
+      if g_file_access.include?(remote_ip) || remote_ip =~ /^63.147.60./
         true
       else
         false
       end
     else
       logger.debug("NON_UFILE ACCESS")
-      # api_response = returns_rights?(context['identifier'])
-      true
+      rights = get_rights(context['identifier'], context['client_ip'])
+      is_not_restricted_for_ip = returns_rights?(rights) && is_not_restricted?(rights, type) #return true if not restricted
     end
   end
 
-  def returns_rights?(image_id)
-    rights = get_rights(image_id)
-    # rough draft of iterpretation of rights statement for restricted images 
+  def derivative_type(size)
+    singleTypeToDimensionMapping = {
+      "2560" => "v",
+      "1600" => "q",
+      "760" => "w",
+      "300" => "r",
+      "150" => "t",
+      "140" => "f",
+      "100" => "b"
+    }

```

Comment:
Because we might get native requests, I think we should assume that sometimes these values won't be exact. I'd recommend instead a range lookup, maybe? 

So like "if size <= 100, it's b. If size <= 140 and > 100 it's f; ... if size > 2560, full res.